server/controllers/chartsController/helpers.js

```diff
@@ -251,41 +256,83 @@ export const postProcessData = (data, studyTotals) => {
   return processedDataBySite
 }
 
-export const mongoQueryFromFilters = (filters) => {
+export const mongoQueriesFromFilters = (filters, userAccess) => {
   if (!filters) {
     return
   }
   const activeFilters = []
-  const facet = {}
+  const facetForFilters = {}
+  const includedCriteriaFacet = {}
   const chrCritFilters = filters.chrcrit_part
     .filter((f) => f.value === TRUE_STRING)
     .map((filter) => FILTER_TO_MONGO_VALUE_MAP[filter.name])
   const includedExcludedFilters = filters.included_excluded
     .filter((f) => f.value === TRUE_STRING)
     .map((filter) => FILTER_TO_MONGO_VALUE_MAP[filter.name])
+  const sexAtBirthFilters = filters.sex_at_birth
+    .filter((f) => f.value === TRUE_STRING)
+    .map((filter) => FILTER_TO_MONGO_VALUE_MAP[filter.name])
 
   if (!!chrCritFilters.length) {
-    facet.chrcrit_part = []
-    facet.chrcrit_part.push({
+    includedCriteriaFacet.chrcrit_part = []
+    includedCriteriaFacet.chrcrit_part.push({
       $match: { chrcrit_part: { $in: chrCritFilters } },
     })
     activeFilters.push('chrcrit_part')
   }
 
   if (!!includedExcludedFilters.length) {
-    facet.included_excluded = []
-    facet.included_excluded.push({
+    includedCriteriaFacet.included_excluded = []
+    includedCriteriaFacet.included_excluded.push({
       $match: { included_excluded: { $in: includedExcludedFilters } },
     })
     activeFilters.push('included_excluded')
   }
 
+  if (!!chrCritFilters.length || !!includedExcludedFilters.length) {
+    facetForFilters.inclusionCriteria = [
+      {
+        $match: {
+          assessment: INCLUSION_EXCLUSION_CRITERIA_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+          inclusionCriteriaCollection: MONGO_COLLECTION_STRING,
+        },
+      },
+    ]
+  }
+
+  if (!!sexAtBirthFilters.length) {
+    facetForFilters.socioDemographics = [
+      {
+        $match: {
+          assessment: SOCIO_DEMOGRAPHICS_FORM,
+          study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
+        },
+      },
+      {
+        $project: {
+          ...INDIVIDUAL_FILTERS_MONGO_PROJECTION,
+          socioDemographicsCollection: MONGO_COLLECTION_STRING,
+        },
+      },
+    ]
+    activeFilters.push('sex_at_birth')
+  }
```

Comment:
what do you think of extracting the logic to create the `facetForFilters` to another function since this one is already really large and we can then test that logic in isolation?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/280/commits/81cad2058d15f6a15a126abcecbb091090e31058